Zone5/Views/ThirdPartyConnectionsView.swift

```diff
@@ -9,52 +9,76 @@
+        case userConnections = "/rest/users/connections"
+        case removeThirdPartyConnection = "/rest/users/connections/rem/{connectionType}"
+        case registerDeviceWithThirdParty = "/rest/users/scheduled/activities/api/v1/push_registrations"
+        case deregisterDeviceWithThirdParty = "/rest/users/scheduled/activities/api/v1/push_registrations/{token}"
+    }
+
+    private let serviceKey: String = "service_name"
+    internal func queryParams(_ serviceType: UserConnectionType) -> URLEncodedBody {
+        let queryParams: URLEncodedBody = [ serviceKey : "\(serviceType)" ]
+        return queryParams
+    }
+
+    /// Register a push token for a device with a 3rd party
+    ///- Parameters:
+    /// - PushRegistration (token, platform, deviceId): Push registration for a device with a third party
+    @discardableResult
+    public func registerDeviceWithThirdParty(registration: PushRegistration, completion: @escaping Zone5.ResultHandler<PushRegistrationResponse>) -> PendingRequest? {
+        return post(Endpoints.registerDeviceWithThirdParty, body: registration, with: completion)
+    }
+
+    /// Deregister a push token for a device with a 3rd party
+    /// - Parameters:
+    /// - token: 3rd party push token to deregister
+    @discardableResult
+    public func deregisterDeviceWithThirdParty(token: String, completion: @escaping Zone5.ResultHandler<Zone5.VoidReply>) -> PendingRequest? {
+        let endpoint = Endpoints.deregisterDeviceWithThirdParty.replacingTokens(["token": token])
+        return delete(endpoint, with: completion)
+    }
+
+    /// Initialize a connection for the current user for the given 3rd party type
+    /// - Parameters
+    @discardableResult
+    public func initializeThirdPartyConnection(type: UserConnectionType, completion: @escaping Zone5.ResultHandler<ThirdPartyInitializeResponse>) -> PendingRequest? {
+        let endpoint = Endpoints.initializePairing.replacingTokens(["connectionType": type.connectionName])
+        return post(endpoint, body: EmptyBody(), with: completion)
+    }
+
+    /// Set an access token for the current user for the given 3rd party type
+    /// - Parameters
+    @discardableResult
+    public func setThirdPartyToken(type: UserConnectionType, parameters: URLEncodedBody?, completion: @escaping Zone5.ResultHandler<Zone5.VoidReply>) -> PendingRequest? {
+        let endpoint = Endpoints.confirmConnection.replacingTokens(["connectionType": type.connectionName])
+        return get(endpoint, parameters: parameters, expectedType: Zone5.VoidReply.self, with: completion)
+    }
+
+    /// Checks if a connection type is enabled or not
+    /// - Parameters
+    @discardableResult
+    public func hasThirdPartyToken(type: UserConnectionType, completion: @escaping Zone5.ResultHandler<Bool>) -> PendingRequest? {
+        return get(Endpoints.userConnections, parameters: queryParams(type), expectedType: [ThirdPartyResponse].self) { result in
+            switch result {
```

Comment:
I don't think this server endpoint takes any query params: https://portal.zone5cloud.com/en/manager/api-documentation/whats-prod/api/api_reference_connections#get-user-connections

Also, since this does not need to call the func queryParams(type) in this class, I don't think this queryParams(type) func is used anywhere anymore and can be removed

Reply:
Done. I don't have access to the documentation.

Reply:
Thanks. You should have access. You should be able to log in with an existing account (like what you would use for Ride), or just sign up a new account. Does that link not work?